exp/controllers/azuremachinepoolmachine_controller.go

```diff
@@ -311,14 +327,10 @@ func (ampmr *AzureMachinePoolMachineController) reconcileDelete(ctx context.Cont
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "controllers.AzureMachinePoolMachineController.reconcileDelete")
 	defer done()
 
+	// Note: If deletion is requested and either the cluster is being deleted or the AMP is missing, we short circuit earlier in the code before calling this function.
+	// This is because machineScope can't be constructed without the cluster or AMP.
 	log.Info("Handling deleted AzureMachinePoolMachine")
 
-	if machineScope.AzureMachinePool == nil || !machineScope.AzureMachinePool.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
This was the previous code to handle when the RG is being deleted. We can't progress with a nil AzureMachinePool in the current implementation so I've moved it to check the cluster scope instead.